src/lib/components/event/event-summary.svelte

```diff
@@ -1,49 +1,104 @@
+  import { page } from '$app/stores';
+  import { eventFilterSort, expandAllEvents } from '$lib/stores/event-view';
+  import { eventHistory } from '$lib/stores/events';
+  import { getPaginatedEvents } from '$lib/services/events-service';
+  import { refresh } from '$lib/stores/workflow-run';
 
-  import Pagination from '$holocene/pagination.svelte';
   import EventSummaryTable from '$lib/components/event/event-summary-table.svelte';
   import EventSummaryRow from '$lib/components/event/event-summary-row.svelte';
   import EventEmptyRow from './event-empty-row.svelte';
-  import Loading from '$lib/holocene/loading.svelte';
-  import { updating } from '$lib/stores/events';
+  import ApiPagination from '$lib/holocene/api-pagination.svelte';
+  import { groupEvents } from '$lib/models/event-groups';
+  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+  import KeyboardShortcut from '$lib/holocene/keyboard-shortcut/shortcut.svelte';
+  import Tooltip from '$lib/holocene/tooltip.svelte';
+  import { authUser } from '$lib/stores/auth-user';
+  import EventShortcutKeys from './event-shortcut-keys.svelte';
 
-  export let items: IterableEvents;
-  export let groups: EventGroups;
-  export let loading = false;
   export let compact = false;
+  let showShortcuts = false;
 
   function handleExpandChange(event: CustomEvent) {
     $expandAllEvents = event.detail.expanded;
   }
+
+  $: category = $page.url.searchParams.get('category');
+  $: total = category ? '' : $eventHistory.end[0]?.id;
+
+  $: fetchEvents = () => {
+    return getPaginatedEvents({
+      namespace: $page.params.namespace,
+      workflowId: $page.params.workflow,
+      runId: $page.params.run,
+      sort: $eventFilterSort,
+      category,
+      compact,
+      settings: $page.stuff.settings,
+      accessToken: $authUser?.accessToken,
+    });
+  };
+
+  const onShiftUp = (event: KeyboardEvent) => {
+    if (!compact) {
+      const sort = 'ascending';
+      $eventFilterSort = sort;
+      updateQueryParameters({
+        parameter: 'sort',
+        value: 'ascending',
+        url: $page.url,
+      });
+    }
+  };
+
+  const onShiftDown = (event: KeyboardEvent) => {
+    if (!compact) {
+      const sort = 'descending';
+      $eventFilterSort = sort;
+      updateQueryParameters({
+        parameter: 'sort',
+        value: sort,
+        url: $page.url,
+      });
+    }
+  };
+
+  const onSpace = (event: KeyboardEvent) => {
+    event.preventDefault();
+    showShortcuts = !showShortcuts;
+  };
 </script>
 
-{#if loading}
-  <Loading />
-{:else}
-  <Pagination
-    {items}
-    floatId="event-view-toggle"
+<EventShortcutKeys open={showShortcuts} {compact} />
+{#key [$eventFilterSort, category, $refresh]}
+  <ApiPagination
     let:visibleItems
-    let:initialItem
+    let:updating
+    let:activeIndex
+    let:setActiveIndex
+    onFetch={fetchEvents}
+    onError={(error) => console.error(error)}
```

Comment:
Is there anything else we want to do with this error?

Reply:
Yes. Totally forgot about this. Good catch.

Reply:
Hot take, we should make our own `log` function that proxies to `console.log` in development but logs to Sentry or _something_ in production.